sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionUtilsMethodHandle.java

```diff
@@ -47,12 +47,12 @@ final class ReflectionUtilsMethodHandle implements ReflectionUtilsApi {
         MethodHandle jdkInternalPrivateLookupInConstructor = null;
 
         try {
+            methodHandlesPrivateLookupIn = lookup.findStatic(MethodHandles.class, "privateLookupIn",
+                MethodType.methodType(MethodHandles.Lookup.class, Class.class, MethodHandles.Lookup.class));
             Class<?> moduleClass = Class.forName("java.lang.Module");
             classGetModule = lookup.unreflect(Class.class.getDeclaredMethod("getModule"));
             moduleIsNamed = lookup.unreflect(moduleClass.getDeclaredMethod("isNamed"));
             moduleAddReads = lookup.unreflect(moduleClass.getDeclaredMethod("addReads", moduleClass));
-            methodHandlesPrivateLookupIn = lookup.findStatic(MethodHandles.class, "privateLookupIn",
-                MethodType.methodType(MethodHandles.Lookup.class, Class.class, MethodHandles.Lookup.class));
```

Comment:
It looks like this can be reverted as you've just moved the code up a few lines

Reply:
This is part of the fix. On Android this try-block raises an exception because there is no Java module support. Moving up this line before the exception occures ensures that at least the privateLookupIn-method is available.

But probably it would be cleaner to move the line to its own try-catch statement.